### DISCUSSION TITLE
"useSigner" not return signer immediately

### DISCUSSION THREAD

**Discussion by minhquan1910** [wevm/wagmi#1169] [Q&A]
My function: 
const { data: wagmiSigner} = useSigner({
    chainId: constants.CHAIN.bscChain.id
});

But when I show the wagmiSigner it is undefined. The signer have value after 2,3 times load the web. How can I handle this problem

**Comment by tmm** [comment]
`useSigner` is asynchronous so won't be returned immediately.
